server/app/views/admin/programs/ProgramImageView.java

```diff
@@ -81,6 +119,70 @@ private FormTag createImageDescriptionForm(
                 .setLabelText("Image description")
                 .setValue(form.value().get().getSummaryImageDescription())
                 .getInputTag())
-        .with(submitButton("Save description").withClass(ButtonStyles.SOLID_BLUE));
+        .with(
+            submitButton("Save description")
+                .withForm(IMAGE_DESCRIPTION_FORM_ID)
+                .withClass(ButtonStyles.SOLID_BLUE));
+  }
+
+  private DivTag createImageUploadForm(ProgramDefinition program) {
+    StorageUploadRequest storageUploadRequest = createStorageUploadRequest(program);
+    FormTag form =
+        fileUploadViewStrategy
+            .renderFileUploadFormElement(storageUploadRequest)
+            .withId(IMAGE_FILE_UPLOAD_FORM_ID);
+    ImmutableList<InputTag> fileUploadFormInputs =
+        fileUploadViewStrategy.fileUploadFormInputs(
+            Optional.of(storageUploadRequest),
+            MIME_TYPES_IMAGES,
+            fileInputId,
+            /* ariaDescribedByIds= */ ImmutableList.of(),
+            /* hasErrors= */ false);
+
+    LabelTag chooseFileButton =
+        label()
+            .withFor(fileInputId)
+            .with(
+                span()
+                    .attr("role", "button")
+                    .attr("tabindex", 0)
+                    .withText("Choose program image")
+                    .withClasses(
+                        ButtonStyles.OUTLINED_TRANSPARENT,
+                        "w-64",
```

Comment:
thought (non-blocking): I don't love the hard-coded width here because it makes it behave differently than other buttons at smaller screen sizes. That said, I can't think of another way to do it, so I think this is fine for now. But I wanted to note it in case you have another idea!

Reply:
Ugh yeah, I definitely agree with you. I'm a bit surprised that removing the width makes the element take the full width instead of just wrapping to fit the content, but maybe that's because it's a `<span>` not a `<div>`?

We have a much better UI coming soon (https://www.figma.com/file/rsQEvXUCuTQwyECapaKzw9/CiviForm---Image-upload-to-program-cards?type=design&node-id=179-6429&mode=design&t=tKf9ygtSP8uRMohP-0), so hoping we can leave this for now since it'll go away soon anyway!

Reply:
That works for me. And ooo that UI looks great!